mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -179,15 +194,21 @@ public EntityDefinition getDefinition(TypeElement classElement) {
   }
 
   private VariableElement findField(
-      TypeElement classElement, String propertyName, TypeMirror fieldType) {
-    for (Element child : classElement.getEnclosedElements()) {
-      if (child.getKind() != ElementKind.FIELD) {
+      Set<TypeElement> typeHierarchy, String propertyName, TypeMirror fieldType) {
+    for (TypeElement classElement : typeHierarchy) {
+      // skip interfaces as they can't have fields
+      if (classElement.getKind().isInterface()) {
         continue;
       }
-      VariableElement field = (VariableElement) child;
-      if (field.getSimpleName().toString().equals(propertyName)
-          && context.getTypeUtils().isSameType(field.asType(), fieldType)) {
-        return field;
+      for (Element child : classElement.getEnclosedElements()) {
+        if (child.getKind() != ElementKind.FIELD) {
+          continue;
+        }
+        VariableElement field = (VariableElement) child;
+        if (field.getSimpleName().toString().equals(propertyName)
```

Comment:
I think `getSimpleName` is what we want, and not sure there is a difference between the qualified and simple name for fields and if there was, simple name is probably preferred since we can scan multiple class for fields.   This is not something I changed (just shows up as a change from nesting things further in code).